src/rgw/rgw_swift.cc

```diff
@@ -683,25 +713,25 @@ int authenticate_temp_url(RGWRados * const store, req_state * const s)
   }
 
   /* Need to try each combination of keys, allowed path and methods. */
-  for (const auto kv : s->user->temp_url_keys) {
+  TempURLSig sig_helper;
+  for (const auto& kv : s->user->temp_url_keys) {
     const int temp_url_key_num = kv.first;
     const string& temp_url_key = kv.second;
 
     if (temp_url_key.empty()) {
       continue;
     }
 
-    for (const auto path : allowed_paths) {
-      for (const auto method : allowed_methods) {
-        const string local_sig = temp_url_gen_sig(temp_url_key,
-                                                  method, path,
-                                                  temp_url_expires);
+    for (const auto& path : allowed_paths) {
+      for (const auto& method : allowed_methods) {
+        const char* local_sig = sig_helper.calc(temp_url_key, method, path,
+						temp_url_expires);
 
         ldout(s->cct, 20) << "temp url signature [" << temp_url_key_num
                           << "] (calculated): " << local_sig
                           << dendl;
 
-        if (local_sig != temp_url_sig) {
+        if (!!sig_helper.comp(temp_url_sig)) {
```

Comment:
I think that renaming the method to `is_equal_to` or similar would better reflect the intetion. Avoding the implicit to-bool conversion through double negation may be another benefit.